ecs-init/config/common.go

```diff
@@ -138,12 +138,18 @@ func DockerUnixSocket() (string, bool) {
 	if dockerHost := os.Getenv("DOCKER_HOST"); strings.HasPrefix(dockerHost, UnixSocketPrefix) {
 		return strings.TrimPrefix(dockerHost, UnixSocketPrefix), true
 	}
-	// return /var/run instead of /var/run/docker.sock, in case the /var/run/docker.sock is deleted and recreated outside the container,
-	// eg: Docker daemon restart
+	// return /var/run instead of /var/run/docker.sock, in case the /var/run/docker.sock is deleted and recreated
+	// outside the container, eg: Docker daemon restart
 	return "/var/run", false
 }
 
 // CgroupMountpoint returns the cgroup mountpoint for the system
 func CgroupMountpoint() string {
 	return cgroupMountpoint
 }
+
+// AgentStdoutStderrLoggingMaxSizeMaxFile returns a tuple of the Maximum log file size and maximum number of log files
+// to be rotated, e.g. 256m, 8
+func AgentStdoutStderrLoggingMaxSize() (string, string) {
```

Comment:
This is headed down the right path, but in order for customers to customize this value they would need to reconfigure this by editing the file and recompiling. There are a few examples in this package that check with the environment variables to determine if the user would like a different setting to be returned. We'll want to give some thought to these variables though, these variables are somewhat unique as they are specific to `ecs-init` itself and we don't have many of them. Let me think a bit on naming and usage.

I prefer that we don't return two different, though closely related, strings from this function. It requires that we remember the order and meaning of each string at the call site. We can avoid the ambiguity by having a function that returns each discrete value (max size and files, so something like `func DockerLogMaxFiles()` and `func DockerLogMaxSize()`) *or* have a function that returns the `godocker.LogConfig{}` struct fully configured.

My vote would be for the multiple function approach, but would be open to consider another if you have suggestions.

Reply:
If we're going for use configurable, the customer MAY want to change their log driver from json-file. Therefore it makes sense to return a godocker.LogConfig{} struct and decide on some environment variables to configure that. I feel deciding on the environment variables is out of scope for this PR, thus I have left that as a TODO item to decide on later.